src/pgo/PGoNetOptions.java

```diff
@@ -195,29 +107,16 @@ public PGoNetOptions(JSONObject config) throws PGoOptionException {
 			}
 
 			JSONObject stateConfig = netConfig.getJSONObject(STATE_FIELD);
-			JSONArray channelsConfig = new JSONArray();
-			int i;
-
-			// try to get the the definition of channels. An algorithm needs not
-			// to define direct channels (perhaps all communication happens
-			// via global variables
-			if (netConfig.has(CHANNELS_FIELD)) {
-				channelsConfig = netConfig.getJSONArray(CHANNELS_FIELD);
-			}
-
 			enabled = true;
 			stateOptions = new StateOptions(stateConfig);
 
-			for (i = 0; i < channelsConfig.length(); i++) {
-				JSONObject channel = (JSONObject) channelsConfig.get(i);
-				String name = channel.getString("name");
-
-				if (channels.get(name) != null) {
-					throw new PGoOptionException("Channel with name \"" + name + "\" already exists");
-				}
-
-				channels.put(name, new Channel(channel));
+			switch (stateOptions.strategy) {
+				default:
```

Comment:
I'd prefer to raise an error when finding an invalid strategy, rather than having etcd as default (i.e., a typo would raise an error instead of silently generating code that is wrong).

Reply:
agreed

Reply:
well, this is checked in StateOptions::validate so logically an invalid strategy will not happen here. However, it's still good to be consistent here.